client.go

```diff
@@ -238,6 +238,26 @@ func (c *Client) Delete(endpoint string) (*Response, error) {
 	})
 }
 
+// DeleteP executes DELETE request to the endpoint with optional query arguments
+//
+// re, err := c.Delete(c.Endpoint("users", "id1"), url.Values{"force": []string{"true"}})
+//
+func (c *Client) DeleteP(endpoint string, params url.Values) (*Response, error) {
+	baseURL, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	baseURL.RawQuery = params.Encode()
+	return c.RoundTrip(func() (*http.Response, error) {
```

Comment:
I think you can simply `c.Delete(baseURL.String())` here instead of duplicating.
And how about `DeleteWithParams`?